skrub/_datetime_encoder.py

```diff
@@ -292,21 +292,21 @@ def _get_datetime_column_indices(X_split, dayfirst=True):
     index_to_format = {}
 
     for col_idx, X_col in enumerate(X_split):
-        X_col = X_col[pd.notnull(X_col)]
-
-        # convert pd.TimeStamp to np.datetime64
-        if all(isinstance(val, pd.Timestamp) for val in X_col):
-            X_col = X_col.astype("datetime64")
+        X_col = X_col[pd.notnull(X_col)]  # X_col is a numpy array
 
         if _is_column_datetime_parsable(X_col):
```

Comment:
I guess if the col is already datetime or Timestamp or numeric -- basically anything else than str or object we can skip this and set datetime_format to None

Reply:
We still want columns with datetime dtype to be added in the datetime index list

Reply:
I meant for datetime columns we don't need to cast to string but anyway it doesn't matter